### DISCUSSION TITLE
Migrating Dapp users from one AA implementation to another AA implementation.

### DISCUSSION THREAD

**Discussion by elallamTaka** [alchemyplatform/aa-sdk#900] [Issues] [answered]
Say that smart accounts were created for a user via ECR-4337 compatible smart contract, deployed via a factory, what is the process of migrating these active users to Alchmeys Modular Smart Accounts?

**Comment by avarobinson** [comment, marked as answer]
Hey @elallamTaka - great question!

You can migrate from one 4337 account implementation to another with one User Operation that will replace the underlying implementation with Modular Account as long as your existing account supports the upgradeToAndCall or upgradeTo methods.

Here are [details](https://accountkit.alchemy.com/resources/faqs#how-can-i-upgrade-from-simple-account-to-light-account) for how to upgrade a Simple Account to Light Account. You'll need to make some changes for Modular Account
- Depending on your existing account implementation, you will need to use the upgradeToAndCall or upgradeTo method
- Instead of Light Account, you need to pass in the Modular Account implementation address to the `newImplementation` param. Find the address depending on chain [here](https://github.com/alchemyplatform/modular-account/tree/develop/deployments)

Considerations:
-  This will maintain the smart contract account address. When using the `createModularAccountAlchemyClient`, you'll want to pass in an `accountAddress` so that a new account is not deployed. 
```
  const smartAccountClient = await createModularAccountAlchemyClient({
    apiKey: API_KEY,
    chain: chain,
    signer: signer,
    gasManagerConfig: {
      policyId: GAS_MANAGER_POLICY_ID,
    },
    accountAddress: "0xACCOUNT_ADDR"
  });
```

What account are you migrating from and which chain? Happy to give more details or answer any other questions!

**Reply by avarobinson** [reply]
@elallamTaka Let me know if this worked for you!

**Reply by elallamTaka** [reply]
Yes, Ill reach out for any question if we start the migration, thanks @avarobinson
